### ISSUE TITLE
What is the future of RaphaelJS? A great SVG editor or a SVG editor limited to VML backward compatibility?

### ISSUE THREAD

**Issue by Soloren2001** [DmitryBaranovskiy/raphael#248]
Dmitry,

I am a really big fan of RaphaelJS. I've been using it as a replacement for Flash.
However, I've been frustrated with some limitations.

These limitations are all due to the fact that initially, RaphaelJS was conceived to allow the use of vector graphics either using SVG or VML for IE users. And this was indeed the biggest problem with vector graphics.

IE 9 is going to be implementing SVG, but surely it will take another 3-7 years before SVG can be used as a stand alone solution since a lot of users will be lagging behind IE9 and more.

In this regard, RaphaelJS is ideally positionned as a great tool during that transition. However, I think that you should also consider the fact that RaphaelJS is more than a "work around' for  cross-browser vector graphics. It's also a great framework and library to create vector graphics.

Therefore, it would be interesting to start to break away from the limitations of functionalities that work only for both VML and SVG. Obviously, the cross browser functionality is still key today but it will slowly fade.

Basically we could imagine have 2 types of functions within RaphaelJS:
- the ones compatible with VML browsers
- the SVG browser only
  So that developpers could choose knowingly what they want to use RaphaelJS.

Ideally, for the "SVG only" functions we could prompt user to update their browser (IE 8 to IE 9, etc...) or to get Chrome Frame or somekind of SVG plugin.

I think this new way to think of RaphaelJS would be quite powerful forward thinking...

I'd love to hear your thoughts on this.

Best regards,

Fabrice
